### PR TITLE
Remove overview AngularJS controllers

### PR DESCRIPTION
**Pull request by yenienserrano**
### Description
This pull request removes the overview AngularJS controllers.
 
### Issues Resolved
- #6593 

### Evidence

image

image

image

### Test

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| The overview must be rendered correctly | :black_circle: | :black_circle: | :black_circle: |
| the navigation to the different modules should work as expected. | :black_circle: | :black_circle: | :black_circle: |
| the `tab` `tabView` `_g` `_a` and if applicable the `agentId` must appear in the URL. | :black_circle: | :black_circle: | :black_circle: |
| Modules must render the corresponding dashboards whether or not they have a pinned agent. | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: The overview must be rendered correctly</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: the navigation to the different modules should work as expected.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: the Tab TabView _g _a and if applicable the agentId must appear in the URL.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Modules must render the corresponding dashboards whether or not they have a pinned agent.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by jbiset — COMMENTED** [review]
## CR: :black_circle: 

## Test UI :green_circle:

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| The overview must be rendered correctly | :green_circle: | :green_circle: | :black_circle: |
| The navigation to the different modules should work as expected and modules must render the corresponding dashboards whether or not they have a pinned agent. | :green_circle: | :green_circle: | :black_circle: |
| the `tab` `tabView` `_g` `_a` and if applicable the `agentId` must appear in the URL. | :green_circle: | :green_circle: | :black_circle: |

**Details**
<details>
<summary>:green_circle: The overview must be rendered correctly</summary>

Chrome - :green_circle:

image

Firefox - :green_circle:

image

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: The navigation to the different modules should work as expected. And modules must render the corresponding dashboards whether or not they have a pinned agent</summary>

Chrome - :green_circle:

[Evidence_1_Chrome.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/beda8dd2-e00b-4658-92f7-87184431c3c0)

Firefox - :green_circle:

[Evidence_1_Firefox.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/cc529832-1673-4588-9a87-decb9f3b6b8e)

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: the Tab TabView _g _a and if applicable the agentId must appear in the URL.</summary>

Chrome - :green_circle:

image

**URL:**

https://0.0.0.0:5601/app/threat-hunting#/overview/?tab=general&tabView=panels&_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-24h,to:now))&_a=(filters:!(('$state':(store:appState),meta:(alias:!n,controlledBy:cluster-manager,disabled:!f,index:'wazuh-alerts-*',key:cluster.name,negate:!f,params:(query:wazuh),removable:!f,type:phrase),query:(match:(cluster.name:(query:wazuh,type:phrase)))),('$state':(store:appState),meta:(alias:!n,controlledBy:pinned-agent,disabled:!f,index:'wazuh-alerts-*',key:agent.id,negate:!f,params:(query:'001'),type:phrase),query:(match:(agent.id:(query:'001',type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:'wazuh-alerts-*',key:agent.name,negate:!f,params:(query:Windows),type:phrase),query:(match_phrase:(agent.name:Windows)))),language:kuery,query:'')&agentId=001

Firefox - :green_circle:

image

**URL:**

https://0.0.0.0:5601/app/threat-hunting#/overview/?tab=general&tabView=panels&_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-24h,to:now))&_a=(filters:!(('$state':(store:appState),meta:(alias:!n,controlledBy:cluster-manager,disabled:!f,index:'wazuh-alerts-*',key:cluster.name,negate:!f,params:(query:wazuh),removable:!f,type:phrase),query:(match:(cluster.name:(query:wazuh,type:phrase)))),('$state':(store:appState),meta:(alias:!n,controlledBy:pinned-agent,disabled:!f,index:'wazuh-alerts-*',key:agent.id,negate:!f,params:(query:'001'),type:phrase),query:(match:(agent.id:(query:'001',type:phrase)))),('$state':(store:appState),meta:(alias:'-%20Level%2012%20or%20above%20alerts',disabled:!f,index:'wazuh-alerts-*',key:query,negate:!f,type:custom,value:'%7B%22bool%22:%7B%22must%22:%5B%5D,%22filter%22:%5B%7B%22bool%22:%7B%22should%22:%5B%7B%22range%22:%7B%22rule.level%22:%7B%22gte%22:12%7D%7D%7D%5D,%22minimum_should_match%22:1%7D%7D%5D,%22should%22:%5B%5D,%22must_not%22:%5B%5D%7D%7D'),query:(bool:(filter:!((bool:(minimum_should_match:1,should:!((range:(rule.level:(gte:12))))))),must:!(),must_not:!(),should:!())))),language:kuery,query:'')&agentId=001

Safari - :black_circle:

</details>

**Review by lucianogorza — COMMENTED** [review]
## Test

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| The overview must be rendered correctly | :green_circle: | :black_circle: | :black_circle: |
| the navigation to the different modules should work as expected. | :black_circle: | :black_circle: | :black_circle: |
| the `tab` `tabView` `_g` `_a` and if applicable the `agentId` must appear in the URL. | :black_circle: | :black_circle: | :black_circle: |
| Modules must render the corresponding dashboards whether or not they have a pinned agent. | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:green_circle: The overview must be rendered correctly</summary>

Chrome - :green_circle:

image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: the navigation to the different modules should work as expected.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: the Tab TabView _g _a and if applicable the agentId must appear in the URL.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Modules must render the corresponding dashboards whether or not they have a pinned agent.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

File: plugins/main/public/components/overview/overview.tsx [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+      )}
+      {tabActive === 'welcome' && (
+        <>
+          <Stats {...agentsCounts} />
+          <OverviewWelcome {...agentsCounts} />
+        </>
+      )}
+    </>
+  );
+};
```
Comment (jbiset) [inline plugins/main/public/components/overview/overview.tsx:223]: I understand that everything related to TabVisualizations, VisFactoryHandler and RawVisualizations would no longer be necessary. No?

**Review by lucianogorza — DISMISSED** [review]

File: plugins/main/public/components/common/welcome/welcome.scss [hunk 2]
```diff
@@ -45,7 +45,7 @@ span.statWithLink:hover {
   text-decoration: underline;
 }
 
-[name="OverviewWelcome"] .flex-redirect-app-links {
```
Comment (Desvelao) [inline plugins/main/public/components/common/welcome/welcome.scss:48]: **question**: there are some remaining styles with the selector `[name="OverviewWelcome"]` on
- plugins/main/public/styles/common.scss
- plugins/main/public/styles/theme/dark/index.dark.scss

Should they be adapted or can be removed?

File: plugins/main/public/controllers/overview/index.js [hunk 3]
```diff
@@ -28,7 +27,6 @@ SelectAgent.displayName = 'SelectAgent';
 RequirementCard.displayName = 'RequirementCard';
 
 app
-  .controller('overviewController', OverviewController)
   .value('OverviewWelcome', OverviewWelcome)
```
Comment (Desvelao) [inline plugins/main/public/controllers/overview/index.js:30]: **nitpick**: this value seems that is not used anymore, we could remove it.

Comment (yenienserrano) [reply plugins/main/public/controllers/overview/index.js]: Solved in https://github.com/wazuh/wazuh-dashboard-plugins/pull/6594/commits/f22e354f610475555fb056b08ea3b5ceca715c91

File: plugins/main/public/components/common/welcome/welcome.scss (hunk 2 above)
Comment (yenienserrano) [reply plugins/main/public/components/common/welcome/welcome.scss:48]: Solved in https://github.com/wazuh/wazuh-dashboard-plugins/pull/6594/commits/f22e354f610475555fb056b08ea3b5ceca715c91

**Review by jbiset — APPROVED** [review]

File: plugins/main/public/controllers/overview/index.js (hunk 3 above)
Comment (Desvelao) [reply plugins/main/public/controllers/overview/index.js]: Regarding the commit, I don't see the `OverviewWelcome` value was removed.

**Review by lucianogorza — APPROVED** [review]

**Review by Desvelao — APPROVED** [review]
The missing changelog entry was added, I will merge it.

**Pull request merged**